### DISCUSSION TITLE
ban time as note in action

### DISCUSSION THREAD

**Discussion by gyterpena** [fail2ban/fail2ban#3148] [Q&A] [answered]
I'm useing <time> as note in actionban (below)

```
actionban = curl -s -X POST "https://api.cloudflare.com/client/v4/zones/bar/firewall/access_rules/rules" \
            -H "Authorization: Bearer foo" \
            -H "Content-Type: application/json" \
            --data '{"mode":"challenge","configuration":{"target":"ip","value":"88.88.88.88"},"notes":"Fail2ban <name> <time>"}'
```

This works ok. 
Does anyone know of a way to convert <time> into human readable format and use it in data portion of curl?

**Comment by sebres** [comment, marked as answer]
do you mean the time?

this is unix time (seconds since posix epoch), so you'd use date to convert it:
```diff
- "... <time>"
+ "... $(date -d "@<time>" --rfc-email)"
```

**Reply by gyterpena** [reply]
Issue is that $(date) gives me bad json as it does not turn into date string when called from --data '' portion of curl. I also tried '$(date)'
`<name> <time>` expands to proper values.

curl -s -X POST "https://api.cloudflare.com/client/v4/zones/bar/firewall/access_rules/rules" \
            -H "Authorization: Bearer foo" \
            -H "Content-Type: application/json" \
            --data '{"mode":"challenge","configuration":{"target":"ip","value":"88.88.88.88"},"notes":"Fail2ban <name> $(date)"}'

**Reply by sebres** [reply]
The whole data is enclosed in single-quotes - this avoids substitution in shell completely.
You need to rewrite it either using double quotes (and escape them inside json too) or with `printf` which can substitute %-tokens:
```bash
$ echo '... $(date)'
... $(date)
$ echo "... $(date)"
... Fri 05 Nov 2021 09:14:26 AM CET
$ echo "$(printf '... %s' "$(date)")"
... Fri 05 Nov 2021 09:15:30 AM CET
```
So try something like this:
```diff
-            --data '{...,"notes":"Fail2ban <name> <time>"}'
+            --data "$(printf '{...,"notes":"Fail2ban %s %s"}' "<name>" "$(date -d "@<time>" --rfc-email)")"
```

**Reply by gyterpena** [reply]
Briant, thanks a lot. Rewriting --data to 
`"{\"mode\":\"challenge\",\"configuration\":{\"target\":\"ip\",\"value\":\"<ip>\"},\"notes\":\"Fail2ban <name> $(date)\"}" `

works.
